pkg/helpers/cmd/errors.go

```diff
@@ -30,3 +32,67 @@ func CheckPodSecurityErr(err error) {
 
 	kcmdutil.CheckErr(err)
 }
+
+const (
+	CustomOIDCReason = "This may occur if an external OIDC provider is configured."
+)
+
+// CheckGroupResourceNotRegisteredErr checks to see if originalError stems from the fact that
+// the GR in question is not registered on the server.
+func CheckGroupResourceNotRegisteredErr(originalError error, discoveryClient discovery.DiscoveryInterface, what, why string) {
+	if agg, ok := originalError.(utilerrors.Aggregate); ok && len(agg.Errors()) == 1 {
+		originalError = agg.Errors()[0]
+	}
+
+	if originalError == nil {
+		return
+	}
+
+	// only NotFound errors can stem from missing GRs, so short-circuit if we're
+	// dealing with any other error
+	if !k8sapierrors.IsNotFound(originalError) {
+		kcmdutil.CheckErr(originalError)
+		return
+	}
+
+	status, ok := originalError.(k8sapierrors.APIStatus)
+	if !ok && !errors.As(originalError, &status) {
+		// this should never happen, since `k8sapierrors.IsNotFound()` succeeded above
+		kcmdutil.CheckErr(originalError)
+		return
+	}
+	details := status.Status().Details
+	if details == nil {
+		// we're not going to be able to detect whether the GR is registered
+		kcmdutil.CheckErr(originalError)
+		return
+	}
+
+	// in the case of a NotFound error, either the GR is not found or the object iself
+	// is not found (when the action was a mutation of existing content); make sure it's
+	// a missing GR by cross-checking discovery
+	_, resourceLists, err := discoveryClient.ServerGroupsAndResources()
```

Comment:
I think we can have errors and still have some data, so we probably want to search the data and ignore the error.

Reply:
Ah, because it's aggregated? Gotcha. Let me do that.